File: example/src/main/java/com/thesurix/example/gesturerecycler/StartActivity.java
package com.thesurix.example.gesturerecycler;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import butterknife.ButterKnife;
import butterknife.OnClick;

public class StartActivity extends AppCompatActivity {

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_start);

        ButterKnife.bind(this);
    }

    @OnClick(R.id.start_grid_btn)
    public void startGridRecycler() {
        startActivity(RecyclerActivity.getIntent(this, RecyclerActivity.RecyclerOption.GRID));
    }

    @OnClick(R.id.start_linear_btn)
    public void startLinearRecycler() {
        startActivity(RecyclerActivity.getIntent(this, RecyclerActivity.RecyclerOption.LINEAR));
    }

    @OnClick(R.id.start_empty_btn)
    public void startLinearEmptyRecycler() {
        startActivity(RecyclerActivity.getIntent(this, RecyclerActivity.RecyclerOption.EMPTY));
    }
}

File: example/src/main/java/com/thesurix/example/gesturerecycler/StartActivity.kt
package com.thesurix.example.gesturerecycler

import android.os.Bundle

import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.thesurix.example.gesturerecycler.databinding.ActivityStartBinding

class StartActivity : AppCompatActivity() {

    private lateinit var binding: ActivityStartBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_start)

        with(binding) {
            startGridBtn.setOnClickListener {
                startRecyclerActivity(RecyclerActivity.RecyclerOption.GRID)
            }

            startLinearBtn.setOnClickListener {
                startRecyclerActivity(RecyclerActivity.RecyclerOption.LINEAR)
            }

            startEmptyBtn.setOnClickListener {
                startRecyclerActivity(RecyclerActivity.RecyclerOption.EMPTY)
            }
        }
    }

    private fun startRecyclerActivity(option: RecyclerActivity.RecyclerOption) {
        startActivity(RecyclerActivity.getIntent(this@StartActivity, option))
    }
}
